.github/workflows/tests.yml

```diff
@@ -41,69 +42,65 @@ jobs:
             conda-version: canary
             test-type: parallel
     env:
-      pytest-replay: --replay-record-dir=pytest-replay/ --replay-base-name=Linux-${{ matrix.conda-version }}-Py${{ matrix.python-version }}
+      CONDA_CHANNEL_LABEL: ${{ matrix.conda-version == 'canary' && 'conda-canary/label/dev' || 'defaults' }}
+      REPLAY_NAME: Linux-${{ matrix.conda-version }}-Py${{ matrix.python-version }}
+      REPLAY_DIR: ${{ github.workspace }}/pytest-replay/
+      PYTEST_MARKER: ${{ matrix.test-type == 'serial' && 'serial' || 'not serial' }}
+      PYTEST_NUMPROCESSES: ${{ matrix.test-type == 'serial' && 0 || 'auto' }}
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
         with:
+          path: ./src/
           fetch-depth: 0
 
-      - name: Setup Miniconda
+      - name: Checkout test recipe
+        uses: actions/checkout@v2
+        with:
+          repository: conda/conda_build_test_recipe
+          path: ./conda_build_test_recipe/
+          fetch-depth: 0
+
+      - name: Setup miniconda
         uses: conda-incubator/setup-miniconda@v2
         with:
-          condarc-file: ci/github/.condarc
+          condarc-file: ./src/ci/github/.condarc
           python-version: ${{ matrix.python-version }}
 
-      - name: Install Conda-Canary if canary version
-        if: matrix.conda-version == 'canary'
-        run: |
-          conda install -y -c conda-canary/label/dev conda
-
-      - name: Install default Conda if release version
-        if: matrix.conda-version == 'release'
+      - name: Setup environment
         run: |
-          conda install -y conda
+          conda install -q -y -c defaults --file tests/requirements.txt
+          conda install -q -y -c defaults --file tests/requirements-linux.txt
+          conda install -q -y ${{ env.CONDA_CHANNEL_LABEL }}::conda
```

Comment:
are teh requirements based on operating system.. wouldn't it be an idea to use an `.env` file that way you can just segragated the installation based on Operating System.. Intead of having multiple `.req.txt` files

Reply:
@mbargull I didn't know we could specify multiple files like that! It also appears that we can also include individual packages at the same time:

```suggestion
          conda install -q -y -c defaults \
              --file tests/requirements.txt \
              --file tests/requirements-linux.txt \
              ${{ env.CONDA_CHANNEL_LABEL }}::conda
```

@sven6002 conda environment files do not support selectors, so we'd still have per OS environments

Reply:
> It also appears that we can also include individual packages at the same time:

Yes, and both ways are fine but have their respective small differences, see the part from my previous comment:
> We could put the `${{ env.CONDA_CHANNEL_LABEL }}::conda` also in the first `install` for the same reasons. A counter argument would be to try to have `${{ env.CONDA_CHANNEL_LABEL }}` out of the channel list when installing the other deps, though. (I added the `-c defaults` here to for config/solver consistency's sake.)

Reply:
🤦🏼‍♂️ lol, I got too excited to finish reading

I see what you mean, well since the conda-canary channel only contains `conda-*` packages and none of these other dependencies it shouldn't make a difference between a one-liner and splitting it?

Reply:
> 🤦🏼‍♂️ lol, I got too excited to finish reading

Hehe :)

> I see what you mean, well since the conda-canary channel only contains `conda-*` packages and none of these other dependencies it shouldn't make a difference between a one-liner and splitting it?

Yes! (I just didn't check what's in -canary these days ;).)